### PR TITLE
[CI] External libraries compilation test

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
## Description of Change
Added you CI workflow for compilation testing External Libraries.
Added documentation page with information how to add a library to the CI test and explanation of result.
Added badge and External Libraries test information to the `README` .
Added `lib_test` label.

Action is running on:
- PR by adding a label `lib_test` adding a comment to PR as a result.
- Schedule every Sunday at 4 am generation a `LIBRARIES_TEST.md` file with the results to the root of repository.

Docs Build Failing -> Fix included in this PR #7906

### CODE REVIEW

File: docs/source/external_libraries_test.rst [hunk 1]
```diff
@@ -0,0 +1,132 @@
+##########################
+External Libraries Testing
+##########################
+
+About
+-----
+
+External libraries testing is a compilation test for listed libraries on arduino-esp32 master branch. The test was made for users, so they can check which libraries are compiling without errors on which SoC.
+External libraries test is running periodicaly (once a week) agains master branch and can also run on PR by adding a label ``lib_test``.
```
Comment (pedrominatel) [inline docs/source/external_libraries_test.rst:9]: "against"

File: docs/source/external_libraries_test.rst [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        "exclude_targets": [],
+        "sketch_path": [
+            "~/Arduino/libraries/ESPAsyncWebServer/examples/CaptivePortal/CaptivePortal.ino",
+            "~/Arduino/libraries/ESPAsyncWebServer/examples/ESP_AsyncFSBrowser/ESP_AsyncFSBrowser.ino",
+            "~/Arduino/libraries/ESPAsyncWebServer/examples/regex_patterns/regex_patterns.ino",
+            "~/Arduino/libraries/ESPAsyncWebServer/examples/simple_server/simple_server.ino"
+        ]
+    },
+
+Sumbit a PR
```
Comment (pedrominatel) [inline docs/source/external_libraries_test.rst:77]: "Submit"

File: docs/source/external_libraries_test.rst [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+***********
+
+* Open a PR with the changes and someone from Espressif team will add a label ``lib_test`` to the PR and CI will run the test to check, if the addition is fine and library / example are compiling.
```
Comment (pedrominatel) [inline docs/source/external_libraries_test.rst:80]: "if the addition is fine and the library/example is compiling."

**Review by pedrominatel — COMMENTED** [review]
Some minor typos.

**Review by pedrominatel — APPROVED** [review]

**Comment by P-R-O-C-H-Y** [comment]
> Some minor typos.

Everything fixed, Thanks for review @pedrominatel.

**Pull request merged**